mne/time_frequency/spectrum.py

```diff
@@ -0,0 +1,1186 @@
+                f'for PSD method "{method}".')
+        self._psd_func = partial(psd_funcs[method], **method_kw)
+
+        # apply proj if desired
+        if proj:
+            inst = inst.copy().apply_proj()
+
+        # prep times and picks
+        self._time_mask = _time_mask(inst.times, tmin, tmax, sfreq=self.sfreq)
+        self._picks = _picks_to_idx(inst.info, picks, 'data',
+                                    with_ref_meg=False)
+
+        # add the info object. bads and non-data channels were dropped by
+        # _picks_to_idx() so we update the info accordingly:
+        self.info = pick_info(inst.info, sel=self._picks, copy=True)
+
+        # assign some attributes
+        self.preload = True  # needed for __getitem__, doesn't mean anything
+        self._method = method
+        self._inst_type = type(inst)
+        # self._dims may also get updated by child classes
+        self._dims = ('channel', 'freq',)
+        if method_kw.get('average', '') in (None, False):
+            self._dims += ('segment',)
+        if method_kw.get('output', '') == 'complex':
+            self._dims = self._dims[:-1] + ('taper',) + self._dims[-1:]
+        # record data type (for repr and html_repr)
+        self._data_type = ('Fourier Coefficients' if 'taper' in self._dims
+                           else 'Power Spectrum')
+
+    def __deepcopy__(self, memodict):
+        """Make a deepcopy."""
+        cls = self.__class__
+        result = cls.__new__(cls)
+        for k, v in self.__dict__.items():
+            v = deepcopy(v, memodict)
+            result.__dict__[k] = v
+        return result
+
+    def __eq__(self, other):
+        """Test equivalence of two Spectrum instances."""
+        # there is one attr (_inst_type) that object_diff can't handle
+        if self._inst_type != other._inst_type:
+            return False
+        _inst_type = self._inst_type
+        del self._inst_type
+        del other._inst_type
+        try:
+            same = object_diff(vars(self), vars(other)) == ''
+        finally:
+            self._inst_type = _inst_type
+            other._inst_type = _inst_type
+        return same
+
+    def __repr__(self):
+        """Build string representation of the Spectrum object."""
+        inst_type_str = self._get_instance_type_string()
+        # shape & dimension names
+        dims = ' × '.join(
+            [f'{dim[0]} {dim[1]}s' for dim in zip(self.shape, self._dims)])
+        freq_range = f'{self.freqs[0]:0.1f}-{self.freqs[-1]:0.1f} Hz'
+        return (f'<{self._data_type} (from {inst_type_str}, '
+                f'{self.method} method) | {dims}, {freq_range}>')
+
+    def _repr_html_(self, caption=None):
+        """Build HTML representation of the Spectrum object."""
+        from ..html_templates import repr_templates_env
+
+        inst_type_str = self._get_instance_type_string()
+        units = ', '.join(
+            f'{ch_type}: {unit}' for ch_type, unit in self.units().items())
```

Comment:
Have you tried `'<br>'.join()` or something similar? I think it would be more readable and less risk of becoming really wide for a single row's data in 

https://output.circle-artifacts.com/output/job/b0cf67fb-c753-4c24-9ec2-0e9877d45f9e/artifacts/0/dev/auto_tutorials/time-freq/10_spectrum_class.html

Reply:
I had the same thought but `'<br/>'.join` did not work (the tag was rendered in the output).  I'll do a bit of searching to see if I can figure out another way to add linebreaks